Modules/Devices/src/Devices.Application/Clients/Commands/CreateClients/Handler.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Clients.Commands.CreateClients;
+
+public class Handler : IRequestHandler<CreateClientCommand, CreateClientResponse>
+{
+    private readonly IOAuthClientsRepository _oAuthClientsRepository;
+
+    public Handler(IOAuthClientsRepository oAuthClientsRepository)
+    {
+        _oAuthClientsRepository = oAuthClientsRepository;
+    }
+
+    public async Task<CreateClientResponse> Handle(CreateClientCommand request, CancellationToken cancellationToken)
+    {
+        if (!string.IsNullOrEmpty(request.ClientId)
+            && await _oAuthClientsRepository.Find(request.ClientId, cancellationToken) != null)
```

Comment:
Why do you read the client if you just use it to check for null? Instead, you could have an "Exists" method on the repository.